internal/handler/msg_usersinfo.go

```diff
@@ -29,6 +38,241 @@ func (h *Handler) MsgUsersInfo(ctx context.Context, msg *wire.OpMsg) (*wire.OpMs
+	}
+
+	var filter *types.Document
+	filter, err = usersInfoQueryFilter(allDBs, singleDB, dbName, users)
+
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	qr, err := usersCol.Query(ctx, &backends.QueryParams{
+		Filter: filter,
+	})
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	defer qr.Iter.Close()
+
+	var res *types.Array
+	res, err = types.NewArray()
+
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	for {
+		_, v, err := qr.Iter.Next()
+
+		if errors.Is(err, iterator.ErrIteratorDone) {
+			break
+		}
+
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		res.Append(v)
+	}
+
+	var reply wire.OpMsg
+	must.NoError(reply.SetSections(wire.OpMsgSection{
+		Documents: []*types.Document{must.NotFail(types.NewDocument(
+			"users", res,
+			"ok", float64(1),
+		))},
+	}))
+
+	return &reply, nil
+}
+
+// usersInfoPair is a pair of username and database name.
+type usersInfoPair struct {
+	username string
+	db       string
+}
+
+// extract username and db from v.
+func (p *usersInfoPair) extract(v any, dbName string) error {
+	switch vt := v.(type) {
+	case *types.Document:
+		ui, err := vt.Get("user")
+		if err != nil {
+			return lazyerrors.Error(err)
+		}
+
+		var ok bool
+		p.username, ok = ui.(string)
+
+		if !ok {
+			return handlererrors.NewCommandErrorMsg(
+				handlererrors.ErrBadValue,
+				fmt.Sprintf("UserName must contain a string field named: user. But, has type %T", ui),
+			)
+		}
+
+		if !vt.Has("db") {
+			return handlererrors.NewCommandErrorMsg(
+				handlererrors.ErrBadValue,
+				"UserName must contain a field named: db",
+			)
+		}
+
+		db, err := vt.Get("db")
+		if err != nil {
+			return lazyerrors.Error(err)
+		}
+
+		if db != nil {
+			p.db, ok = db.(string)
+			if !ok {
+				return handlererrors.NewCommandErrorMsg(
+					handlererrors.ErrBadValue,
+					fmt.Sprintf("UserName must contain a string field named: db. But, has type %T", db),
+				)
+			}
+		}
+
+		return nil
+	case string:
+		p.username = vt
+		p.db = dbName
+
+		return nil
+	default:
+		return lazyerrors.Errorf("unexpected type %T", vt)
+	}
+}
+
+// usersInfoQueryFilter returns a filter for usersInfo command.
+//
+// When allDBs is true, it returns a filter for all databases.
+// When singleDB is true, it returns a filter for a single database (case when usersInfo: 1 is invoked).
+// Otherwise, it filters by any pair of user and database.
+func usersInfoQueryFilter(allDBs, singleDB bool, dbName string, pairs []usersInfoPair) (*types.Document, error) {
+	filter := must.NotFail(types.NewDocument())
+
+	if allDBs {
+		return filter, nil
+	}
+
+	if singleDB {
+		filter.Set("db", dbName)
+		return filter, nil
+	}
+
+	ps := []any{}
+	for _, p := range pairs {
+		ps = append(ps, p.db+"."+p.username)
+	}
+
+	ids, err := types.NewArray(ps...)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	filter.Set("_id", must.NotFail(types.NewDocument(bson.D{{Key: "$eq", Value: ids}})))
```

Comment:
I guess you need smth like 

```suggestion
	filter.Set("_id", must.NotFail(types.NewDocument("$eq", ids)))
```